frontend/lib/search/search_page.dart

```diff
@@ -18,11 +22,25 @@ class _SearchPageState extends State<SearchPage> {
   String? searchFieldText;
   final List<CategoryAsset> _selectedCategories = [];
   CoordinatesInput? _coordinates;
+  SortingMode? _sortingMode;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      final settings = context.read<SettingsModel>();
+      setState(() => _sortingMode = settings.locationFeatureEnabled && _coordinates != null
+          ? SortingMode.byDistance
+          : SortingMode.alphabetically);
+    });
+  }
 
   @override
   Widget build(BuildContext context) {
     final t = context.t;
-    final currentCoordinatesInput = _coordinates;
+    if (_sortingMode == null) {
```

Comment:
I would rather prefer to set  a default value for the sorting mode and make it mandatory instead of checking here for null value. Even you could remove this "!" for `sortingMode`
`SortingMode _sortingMode = SortingMode.alphabetically;`

Reply:
The reason why I did not opt for a default value is that this might lead to flickering, as the sorting mode is automatically updated/changed at a later point if the location feature is enabled and user coordinates are available.

Reply:
i couldn't recognize a flickering. but i'm also fine to keep your solution